apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Tests/SubscriberClientTest.cs

```diff
@@ -953,5 +1010,77 @@ public void DeliveryAttempt()
                 });
             }
         }
+
+        // Acknowledge / ModifyAcknowledgeDeadline calls may throw RpcException. RpcExceptions should not be thrown to the client.
+        [Theory, CombinatorialData]
+        public void AckModifyAckDeadlineFault_NotThrown(
+            [CombinatorialValues(true, false, null)] bool? ackOrModifyAck)
+        {
+            var msgs = new[] {
+                ServerAction.Data(TimeSpan.Zero, new[] { "1" }),
+                ServerAction.Data(TimeSpan.FromSeconds(5), new[] { "2" }),
+                ServerAction.Data(TimeSpan.Zero, new[] { "3" }),
+                ServerAction.Data(TimeSpan.Zero, new[] { "4" }),
+                ServerAction.Inf()
+            };
+
+            // Any RpcException thrown from Acknowledge and/ or ModifyAcknowledgementDeadline should not be thrown to the client.
+            var rpcException = new RpcException(new Status(StatusCode.InvalidArgument, ""), "");
+            // If ackOrModifyAck is true, then Acknowledge (Ack) will throw the supplied RpcException 2 times.
+            // If ackOrModifyAck is false, then ModifyAcknowledgementDeadline (ModifyAck) will throw the supplied RpcException 2 times.
+            // If ackOrModifyAck is null, then both Ack and ModifyAck will throw the supplied RpcException, total 2 times.
+            var ackModifyAckDeadlineAction = ackOrModifyAck == null ? AckModifyAckDeadlineAction.Data(rpcException, 2, true, true) : ackOrModifyAck.Value ? AckModifyAckDeadlineAction.BadAck(rpcException, 2) : AckModifyAckDeadlineAction.BadModifyAckDeadline(rpcException, 2);
```

Comment:
```suggestion
            var ackModifyAckDeadlineAction =
                ackOrModifyAck == null ? AckModifyAckDeadlineAction.Data(rpcException, 2, true, true)
                : ackOrModifyAck.Value ? AckModifyAckDeadlineAction.BadAck(rpcException, 2)
                : AckModifyAckDeadlineAction.BadModifyAckDeadline(rpcException, 2);
```

(That's the way we normally format this sort of multiple-conditional operator.)